pkg/kexecbin/kexec_bin.go

```diff
@@ -0,0 +1,74 @@
+package kexecbin
+
+import (
+	"errors"
+	"os"
+	"os/exec"
+	"runtime"
+)
+
+var (
+	// FirstDTBPath is the first dtp path to check
+	FirstDTBPath = "/sys/firmware/fdt"
+	// SecondDTBPath is the second dtp path to check
+	SecondDTBPath = "/proc/device-tree"
+)
+
+// KexecBin uses kexec-tools binary and runtime architecture detection
+// to execute abritary files.
+func KexecBin(kernelFilePath string, kernelCommandline string, initrdFilePath string, dtFilePath string) error {
+	var err error
+	var baseCmd string
+	switch runtime.GOARCH {
```

Comment:
You can do this at compile time:

- Have a global variable `var kexecPath string`
- In the file "kexecbin_amd64.go", have `func init { kexecPath = "kexec-amd64" }`
- In the file "kexecbin_arm64.go", have `func init { kexecPath = "kexec-arm64" }`

Reply:
remove it. We just searching for kexec and during u-root initramfs generation users have to pass the right binary